version: '3.9'

volumes:
  postgres:
  rabbitmq:

services:

  postgres:
    image: postgres:16-alpine3.19
    environment:
      POSTGRES_PASSWORD: postgres
    ports:
      - "0.0.0.0:5432:5432"
    networks:
      - dev-network
    volumes:
      - postgres:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "PGPASSWORD=postgres pg_isready -U postgres -h postgres"]
      interval: 10s
      timeout: 30s
      retries: 15

  rabbitmq:
    image: rabbitmq:3.4-management
    environment:
      RABBITMQ_DEFAULT_USER: rabbitmq
      RABBITMQ_DEFAULT_PASS: rabbitmq
    ports:
      - "0.0.0.0:5672:5672"
      - "0.0.0.0:15672:15672"
    networks:
      - dev-network
    volumes:
      - rabbitmq:/var/lib/rabbitmq
    healthcheck:
      test: rabbitmq-diagnostics -q ping
      interval: 10s
      timeout: 30s
      retries: 15

  redis:
    image: redis:7-alpine
    ports:
      - "0.0.0.0:6379:6379"
    networks:
      - dev-network
    healthcheck:
      test: redis-cli ping
      interval: 10s
      timeout: 30s
      retries: 15

  jaeger:
    image: jaegertracing/all-in-one:1.55
    environment:
      - COLLECTOR_OTLP_ENABLED=true
      - LOG_LEVEL=debug
    ports:
      - "0.0.0.0:16686:16686"
      - "0.0.0.0:4317:4317"
    networks:
      - dev-network

networks:
  dev-network:
